wolfcrypt/src/rsa.c

```diff
@@ -3147,6 +3172,23 @@ static int wc_RsaFunction_ex(const byte* in, word32 inLen, byte* out,
 #else /* !WOLF_CRYPTO_CB_ONLY_RSA */
     SAVE_VECTOR_REGISTERS(return _svr_ret;);
 
+    /* perform software padding here */
+    if(!padded) {
+        in2Len = inLen;
+        in2 = in;
+        if ((type == RSA_PUBLIC_ENCRYPT || type == RSA_PRIVATE_ENCRYPT) && padding->pad_type != WC_RSA_NO_PAD) {
```

Comment:
```[clang-tidy-defaults] [7 of 7] [wolfssl]
    configure...   real 0m7.540s  user 0m4.942s  sys 0m2.493s
    build.../tmp/workspace/PRB-multi-test-script/wolfssl/wolfcrypt/src/rsa.c:3179:94: warning: The left operand of '!=' is a garbage value [clang-analyzer-core.UndefinedBinaryOperatorResult]
if ((type == RSA_PUBLIC_ENCRYPT || type == RSA_PRIVATE_ENCRYPT) && padding->pad_type != WC_RSA_NO_PAD) {
^
/tmp/workspace/PRB-multi-test-script/wolfssl/wolfcrypt/src/rsa.c:4288:12: note: Calling 'wc_RsaPSS_Sign_ex'
return wc_RsaPSS_Sign_ex(in, inLen, out, outLen, hash, mgf,
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~```

Reply:
Hmmm - can't seem to workaround this one - seems like a false positive of my version of scan-build.